frontend/src/pages/team/Devices/index.vue

```diff
@@ -160,35 +160,44 @@ export default {
                 this.devices[index] = device
             }
         },
-        async deleteDevice (device) {
-            await deviceApi.deleteDevice(device.id)
-            const index = this.devices.indexOf(device)
-            this.devices.splice(index, 1)
-        },
         async assignDevice (device, projectId) {
             const updatedDevice = await deviceApi.updateDevice(device.id, { project: projectId })
             device.project = updatedDevice.project
         },
-        async unassignDevice (device) {
-            await deviceApi.updateDevice(device.id, { project: null })
-            delete device.project
-            // If this component is being used on the Team/Device page - this unassign should
-            // remove it from the view.
-            if (this.isProjectDeviceView) {
-                const index = this.devices.indexOf(device)
-                this.devices.splice(index, 1)
-            }
-        },
         deviceAction (action, deviceId) {
             const device = this.devices.find(d => d.id === deviceId)
             if (action === 'edit') {
                 this.showEditDeviceDialog(device)
             } else if (action === 'delete') {
-                this.$refs.confirmDeviceDeleteDialog.show(device)
+                Dialog.show({
+                    header: 'Delete Device',
+                    kind: 'danger',
+                    text: 'Are you sure you want to delete this device? Once deleted, there is no going back.',
+                    confirmLabel: 'Delete'
+                }, async () => {
+                    await deviceApi.deleteDevice(device.id)
+                    const index = this.devices.indexOf(device)
+                    this.devices.splice(index, 1)
+                })
             } else if (action === 'updateCredentials') {
                 this.$refs.deviceCredentialsDialog.show(device)
```

Comment:
Can this not be replaced with `Dialog.Show`? It opens this...
![image](https://user-images.githubusercontent.com/44235289/185444936-924b77c6-5f64-4029-b900-1efc76ac9bff.png)

Reply:
Good spot, I missed that one. If you'd like to try and build it yourself, could be a good exercise to try out the API? Also happy to do it myself of course.

Reply:
@joepavitt can you pick up this last one please?

Reply:
Having taken a look, whilst that condition looks simple on the surface, actually looking into the [dialog itself](https://github.com/flowforge/flowforge/blob/dialog-api/frontend/src/pages/team/Devices/dialogs/DeviceCredentialsDialog.vue) it does have a lot of complex logic in. We could extract the "Are you sure" into the JS API, but it is tightly coupled with the credentials generation dialog, which in itself would still need to stay as a standalone component

Reply:
ok, NP.